galaxy/tools/deps/mulled/mulled_search.py

```diff
@@ -76,57 +92,310 @@ def search_repository(self, search_string, non_strict):
                     results_tmp = searcher.search(query)
                     results.extend(results_tmp)
 
-            sys.stdout.write("The query \033[1m %s \033[0m resulted in %s result(s).\n" % (search_string, len(results)))
-
-            if non_strict:
-                sys.stdout.write('The search was relaxed and the following search terms were searched: ')
-                sys.stdout.write('\033[1m %s \033[0m\n' % ', '.join(suggestions))
-
             out = list()
+
             for result in results:
                 title = result['title']
                 for version in self.get_additional_repository_information(title):
-                    row = [title]
-                    row.append(version)
-                    out.append(row)
-            if out:
-                col_width = max(len(word) for row in out for word in row) + 2  # padding
-                for row in out:
-                    name = row[0]
-                    version = row[1]
-                    sys.stdout.write("".join(word.ljust(col_width) for word in row) + "docker pull quay.io/%s/%s:%s\n" % (self.organization, name, version))
-            else:
-                sys.stdout.write("No results found for %s in quay.io/%s.\n" % (search_string, self.organization))
+                    out.append({'package': title, 'version': version,})
+
+            return out
 
     def get_additional_repository_information(self, repository_string):
         """
         Function downloads additional information from quay.io to
         get the tag-field which includes the version number.
         """
         url = "%s/%s/%s" % (QUAY_API_URL, self.organization, repository_string)
-        r = requests.get(url, headers={'Accept-encoding': 'gzip'}, timeout=12)
+        r = requests.get(url, headers={'Accept-encoding': 'gzip'})
 
         json_decoder = json.JSONDecoder()
         decoded_request = json_decoder.decode(r.text)
         return decoded_request['tags']
 
+class CondaSearch():
+    """
+    Tool to search the bioconda channel
```

Comment:
Not only bioconda please, make bioconda default, but provide a way to overwrite this.